Sources/Turf/WKT.swift

```diff
@@ -0,0 +1,222 @@
+            print("error: Failed to init 'MultiPoint' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension LineString: WKTCodable {
+    public var WKTString: String {
+        return "LINESTRING\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [LineString])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'LineString' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension MultiLineString: WKTCodable {
+    public var WKTString: String {
+        return "MULTILINESTRING\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [MultiLineString])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'MultiLineString' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension Polygon: WKTCodable {
+    public var WKTString: String {
+        return "POLYGON\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [Polygon])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'Polygon' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension MultiPolygon: WKTCodable {
+    public var WKTString: String {
+        return "MULTIPOLYGON\(coordinates.wktCoordinatesString)"
+    }
+    
+    public init?(fromWKT wkt: String) {
+        do {
+            let data = try WKTParser.parse(wkt)
+            guard let object = (data as? [MultiPolygon])?.first else {
+                return nil
+            }
+            self = object
+        } catch {
+            print("error: Failed to init 'MultiPolygon' from WKT with error: \(error)")
+            return nil
+        }
+    }
+}
+
+extension Geometry: WKTCodable {
+    public var WKTString: String {
+        switch self {
+        case .point(let geometry):
+            return geometry.WKTString
+        case .lineString(let geometry):
+            return geometry.WKTString
+        case .polygon(let geometry):
+            return geometry.WKTString
+        case .multiPoint(let geometry):
+            return geometry.WKTString
+        case .multiLineString(let geometry):
+            return geometry.WKTString
+        case .multiPolygon(let geometry):
+            return geometry.WKTString
+        case .geometryCollection(let geometry):
+            return geometry.WKTString
```

Comment:
We didn’t end up creating a bottleneck method like this for `GeometryConvertible.geometry`, so should we do it for `WKTCodable.WKTString`? Or is the rationale for this bottleneck method that decoding to WKT shouldn’t require using a `Decoder`?

Reply:
`Decoder` is indeed not a preferable way here (IMHO) for several reasons. There is no clear path or guide on how to write custom Coders and Decoders. Related publicly visible protocols are not meant to be guide on that process, and majority of logic for existing JSONDecoder or PropertyListDecoder are hidden. Also, this would potentially mean re-using `encode/decode` methods, but since WKT and JSON formats are too different (in terms of syntax and data content) this would overcomplicate the methods. Plus, there should be some trick required to distinguish which type of coder is used for each particular call.
And at last our use-case is actually not too complex, so I decided we could get away with simple method which would convert object to a string. There is no wild nesting or huge variety of types and data that could be inside geometries, so full scale Coder may be an overkill.

Reply:
I agree about avoiding Coder, although I meant to ask if we really needed `Geometry.WKTString` as a bottleneck method. On second thought, this method is fine. 👍 Unlike `GeometryConvertible.geometry`, `WKTConvertible.wktString` always returns a `String`, so there’s no benefit to making callers perform this switch themselves. Also, the `Geometry` type only serves a fixed number of geometry types; any other `GeometryConvertible` type would be irrelevant here.